### PR TITLE
Halo2 shplonk  bench

### PR DESCRIPTION
**Pull request by tyshko-rostyslav**
Changed to uniform benchmark representation

### CODE REVIEW

**Review by jonesmarvin8 — APPROVED** [review]
Approved. No changes necessary.

File: halo2-shplonk/benches/poseidon.rs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        let snarks: Vec<_> = vec![0; k].into_iter().map(|_| halo2_shplonk::gen_application_snark(&params_app)).collect();
+
+        let agg_circuit = AggregationCircuit::<SHPLONK>::new(&params, snarks);
+        let pk = gen_pk(
+            &params,
+            &agg_circuit.without_witnesses(),
+            None,
+        );
+
+        snark_verifier_sdk::halo2::gen_proof_gwc(
```
Comment (jonesmarvin8) [inline halo2-shplonk/benches/poseidon.rs:45]: gen_proof_shplonk

**Review by jonesmarvin8 — CHANGES_REQUESTED** [review]

File: halo2-shplonk/benches/poseidon.rs (hunk 1 above)
Comment (jonesmarvin8) [reply halo2-shplonk/benches/poseidon.rs:45]: Also: to match with other benchmarks like Arecibo and the two nova's, there probably should be a dedicated verify benchmark.

Comment (tyshko-rostyslav) [reply halo2-shplonk/benches/poseidon.rs]: Done

**Review by jonesmarvin8 — APPROVED** [review]
Approved.

**Pull request merged**